sys-devel/bpf-toolchain/bpf-toolchain-20240407.ebuild

```diff
@@ -0,0 +1,196 @@
+	mv gcc{-${GCC_PV},} || die
+
+	default
+}
+
+src_compile() {
+	# not great but do everything in src_compile given bootstrapping
+	# process needs to be done in steps of configure+compile+install
+	# (done modular to have most package-specific things in one place)
+
+	CTARGET=bpf-unknown-none
+
+	BPFT_D=${T}/root # moved to ${D} in src_install
+	local prefix=${EPREFIX}/usr
+	local sysroot=${BPFT_D}${prefix}
+	local -x PATH=${sysroot}/bin:${PATH}
+
+	use custom-cflags || strip-flags # fancy flags are not realistic here
+
+	local multilib=false
+
+	# global configure flags
+	local conf=(
+		--build=${CBUILD:-${CHOST}}
+		--target=${CTARGET}
+		--{doc,info,man}dir=/.skip # let the real binutils+gcc handle docs
+		MAKEINFO=: #922230
+	)
+
+	# binutils
+	local conf_binutils=(
+		--prefix="${prefix}"
+		--host=${CHOST}
+		--disable-cet
+		--disable-default-execstack
+		--disable-nls
+		--disable-shared
+		--with-system-zlib
+		--without-debuginfod
+		--without-msgpack
+		--without-zstd
+	)
+
+	# gcc (minimal -- if need more, disable only in stage1 / enable in stage3)
+	local conf_gcc=(
+		--prefix="${prefix}"
+		--host=${CHOST}
+		--disable-bootstrap
+		--disable-cc1
+		--disable-cet
+		--disable-gcov #843989
+		--disable-gomp
+		--disable-nls # filename collisions
+		--disable-libcc1
+		--disable-libquadmath
+		--disable-libsanitizer
+		--disable-libssp
+		--disable-libvtv
+		--disable-shared
+		--disable-werror
+		--enable-languages=c
+		--with-gcc-major-version-only
+		--with-system-zlib
+		--without-isl
+		--without-zstd
+		--disable-multilib
+	)
+
+	# libstdc++ may misdetect sys/sdt.h on systemtap-enabled system and fail
+	# (not passed in conf_gcc above given it is lost in sub-configure calls)
+	local -x glibcxx_cv_sys_sdt_h=no
+
+	# bpft-build <path/package-name>
+	# -> ./configure && make && make install && bpft-package()
+	# passes conf and conf_package to configure, and users can add options
+	# through environment with e.g.
+	#	BPFT_BINUTILS_CONF="--some-option"
+	#	EXTRA_ECONF="--global-option" (generic naming for if not reading this)
+	bpft-build() {
+		local id=${1##*/}
+		local build_dir=${WORKDIR}/${1}-build
+
+		# econf is not allowed in src_compile and its defaults are
+		# mostly unused here, so use configure directly
+		local conf=( "${WORKDIR}/${1}"/configure "${conf[@]}" )
+
+		local -n conf_id=conf_${id}
+		[[ ${conf_id@a} == *a* ]] && conf+=( "${conf_id[@]}" )
+
+		local -n extra_id=BPFT_${id^^}_CONF
+		conf+=( ${EXTRA_ECONF} ${extra_id} )
+
+		einfo "Building ${id} in ${build_dir} ..."
+
+		mkdir -p "${build_dir}" || die
+		pushd "${build_dir}" >/dev/null || die
+
+		edo "${conf[@]}"
+		emake MAKEINFO=: V=1
+		# -j1 to match bug #906155, other packages may be fragile too
+		emake -j1 MAKEINFO=: V=1 DESTDIR="${BPFT_D}" install
+
+		declare -f bpft-${id} >/dev/null && edo bpft-${id}
+
+		popd >/dev/null || die
+	}
+
+	# build with same ordering that crossdev would do
+	bpft-build binutils
+	bpft-build gcc
```

Comment:
If this doesn't need multi-stage bootstrapping, is it really necessary to be combined in 1 ebuild? Was under the impression the process was going to be more involved.

Arguably mingw64-toolchain could've split out binutils too, but still had to combine w/ runtime for the multi-stage builds.

Reply:
(not that I think of it as a problem if just want to keep things in one place in a similar manner)